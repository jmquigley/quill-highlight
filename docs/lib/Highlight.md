<a name="module_Highlight"></a>

## Highlight
A custom Quill highlighting module.  It manages multiple mode instances
that are used to apply text formatting to the contents of the control.

The module is added to the component per the API instructions:
https://quilljs.com/guides/building-a-custom-module/

Once the module is initialized the highlighting mode can be changed with
the `set` function.  This just changes what the reference internally to
another class that handles the formatting.

#### Examples:

```javascript
import {EditorMode, EditorStyle, Highlight} from './modules';

Quill.register('modules/highlight', Highlight);
const quill = new Quill('#editor', {
    formats: '',
    modules: {
        highlight: {
        }
    },
    theme: 'snow'
});
...
const highlight = quill.getModule('highlight');
highlight.set({
    content: 'some value',
    mode: EditorMode.markdown,
    styling: EditorStyle.monokai
});
```

