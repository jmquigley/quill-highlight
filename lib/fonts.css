@font-face {
	font-family: 'Fira Code';
	src: local('FiraCode'), local('FiraCode-Regular'), url('./fonts/FiraCode.ttf');
}

@font-face {
	font-family: 'Inconsolata';
	src: local('Inconsolata'), local('Inconsolata-Regular'), url('./fonts/Inconsolata.ttf');
}

@font-face {
	font-family: 'Source Code Pro';
	src: local('SourceCodePro'), local('SourceCodePro-Regular') url('./fonts/SourceCodePro.ttf');
}
