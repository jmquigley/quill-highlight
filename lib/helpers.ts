// This will get a module level import of a Quill instance, but it will not
// be the one on the window.
//
// globals suck.
//
// import * as q from 'quill';

const debug = require("debug")("quill-markup.helpers");

/**
 * Retrieves the global instance of Quill.  When Quill is used in a browser, it
 * is loaded with the `<script>` tag.  This places the global instance on
 * "global" (or window).  There must only be one copy of this instance.  If you use
 * "require" or import to pull it in it will be scoped to that module and will
 * not be the same as the global version.  Bad things happen when one does this
 * as custom, registered modules and formats will not work unless one registers
 * them on the correct instance that is shown in the browser.   This check and
 * retrieval will ensure that it is uses the global instance.
 * @returns {Quill} the reference to the global Quill instance
 */
export function getQuill() {
	const gQuill = (global as any).Quill;
	if (!gQuill) {
		throw new Error("Quill is not loaded in this application");
	}

	debug("Retrieving global quill instance: %O", gQuill);
	return gQuill;
}

const Quill = getQuill();
const Delta = Quill.import("delta");
const Parchment = Quill.import("parchment");

export {Delta, Parchment, Quill};
