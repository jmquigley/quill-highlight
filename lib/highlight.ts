/**
 * A custom Quill highlighting module.  It manages multiple mode instances
 * that are used to apply text formatting to the contents of the control.
 *
 * The module is added to the component per the API instructions:
 * https://quilljs.com/guides/building-a-custom-module/
 *
 * Once the module is initialized the highlighting mode can be changed with
 * the `set` function.  This just changes what the reference internally to
 * another class that handles the formatting.
 *
 * #### Examples:
 *
 * ```javascript
 * import {EditorMode, EditorStyle, Highlight} from './modules';
 *
 * Quill.register('modules/highlight', Highlight);
 * const quill = new Quill('#editor', {
 *     formats: '',
 *     modules: {
 *         highlight: {
 *         }
 *     },
 *     theme: 'snow'
 * });
 * ...
 * const highlight = quill.getModule('highlight');
 * highlight.set({
 *     content: 'some value',
 *     mode: EditorMode.markdown,
 *     styling: EditorStyle.monokai
 * });
 * ```
 *
 * @module Highlight
 */

'use strict';

import {BaseHighlightMode, Markdown, Text} from './modes';

export enum HighlightMode {
	markdown,
	richedit,
	text
}

export enum HighlightStyle {
	custom = 'custom',
	plain = 'plain',
	monokai = 'monokai'
}

export interface HighlightOptions {
	content?: string;
	custom?: any;
	mode: HighlightMode;
	styling: HighlightStyle;
}

const debug = require('debug')('highlight');

export class Highlight {

	private _modes: Map<HighlightMode, any> = new Map<HighlightMode, any>();
	private _opts: HighlightOptions;
	private _processor: BaseHighlightMode;
	private _quill: any;
	private _styles: Map<string, any> = new Map<string, any>();

	constructor(quill: any, opts: HighlightOptions) {
		debug('Initializing highlight module');

		this._quill = quill;

		this._styles[HighlightStyle.monokai] = require('./styles/monokai.json');
		this._modes[HighlightMode.markdown] = new Markdown(quill);
		this._modes[HighlightMode.text] = new Text(quill);
		this.handleChange = this.handleChange.bind(this);
		quill.on('text-change', this.handleChange);
		this.set(opts);
	}

	public set(opts: HighlightOptions) {

		this._opts = opts = Object.assign({
			mode: HighlightMode.text,
			styling: HighlightStyle.plain,
			custom: {},
			content: ''
		}, opts);

		this._processor = this._modes[opts.mode];
		this._processor.content = opts.content;
		this._processor.style = this._styles[opts.styling];

		debug(`initial content: ${opts.content}`);
		this._quill.setText(opts.content);
	}

	private handleChange(delta: any, old: any, source: string) {
		debug(`highlight change: %o, old: %o, source: %s`, delta, old, source);

		const text: string = this._quill.getText();
		this._processor.highlight(text, 0, text.length);
	}
}
