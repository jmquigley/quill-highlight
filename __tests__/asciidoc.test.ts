"use strict";

import * as fs from "fs-extra";
import * as path from "path";
// import * as sinon from 'sinon';
import {Fixture} from "util.fixture";
import {join} from "util.join";
import {cleanup} from "./helpers";

const debug = require("debug")("asciidoc.test");

// can't use this before the global require and jsdom initialization
import {Quill} from "../lib/helpers";
let quill: any = null;

import {Markup, MarkupMode} from "../index";

afterAll((done) => {
	cleanup(path.basename(__filename), done);
});

beforeEach(() => {
	const data = fs
		.readFileSync(join(__dirname, "fixtures", "empty-html", "index.html"))
		.toString("utf8");

	document.body.innerHTML = data;
	quill = new Quill("#editor", {
		theme: "snow"
	});

	expect(quill).toBeDefined();
});

test("Create Markup instance with Asciidoc mode", () => {
	const fixture = new Fixture("asciidoc");
	expect(fixture).toBeDefined();

	const txt = fixture.read("file.txt");
	expect(txt).toBeDefined();

	const markup = new Markup(quill, {
		content: txt,
		mode: MarkupMode.asciidoc
	});

	expect(markup).toBeDefined();
	expect(markup.quill).toBeDefined();
	expect(markup.editor).toBeDefined();

	markup.refresh();

	const delta = markup.quill.getContents();
	expect(delta).toBeDefined();
	debug("%j", delta);

	expect(delta).toMatchSnapshot();
});

test("Use markup set call to change the mode to asciidoc", () => {
	const markup = new Markup(quill);

	expect(markup).toBeDefined();
	expect(markup.quill).toBeDefined();
	expect(markup.editor).toBeDefined();

	markup.set({
		content: "test",
		custom: {
			background: "red",
			foreground: "yellow"
		},
		fontName: "Consolas",
		fontSize: 14,
		mode: MarkupMode.asciidoc
	});

	expect(markup.opts).toBeDefined();
	expect(markup.opts).toMatchSnapshot();
});

test("Use markup bold call with Asciidoc", () => {
	const markup = new Markup(quill, {
		content: "test",
		mode: MarkupMode.asciidoc
	});

	expect(markup).toBeDefined();
	expect(markup.quill).toBeDefined();
	expect(markup.editor).toBeDefined();

	markup.setBold();

	const delta = markup.quill.getContents();
	expect(delta).toBeDefined();
	debug("%j", delta);

	expect(delta).toMatchSnapshot();
});

for (const level of ["0", "1", "2", "3", "4", "5", "6"]) {
	test(`Use markup header ${level} call with Asciidoc`, () => {
		const markup = new Markup(quill, {
			content: "test",
			mode: MarkupMode.asciidoc
		});

		expect(markup).toBeDefined();
		expect(markup.quill).toBeDefined();
		expect(markup.editor).toBeDefined();

		markup.setHeader(level);

		const delta = markup.quill.getContents();
		expect(delta).toBeDefined();
		expect(delta).toMatchSnapshot();
	});
}
