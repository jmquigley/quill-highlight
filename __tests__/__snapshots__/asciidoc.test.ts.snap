// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Markup instance with Asciidoc mode 1`] = `
Delta {
  "ops": Array [
    Object {
      "attributes": Object {
        "color": "",
      },
      "insert": ":tags:",
    },
    Object {
      "insert": " test, test, test

some .

",
    },
    Object {
      "attributes": Object {
        "color": "#ffff00",
      },
      "insert": "test",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#ffff00",
      },
      "insert": "====",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "#ffd700",
      },
      "insert": "test",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#ffd700",
      },
      "insert": "----",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "#ffaf00",
      },
      "insert": "test",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#ffaf00",
      },
      "insert": "~~~~",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "#ff8700",
      },
      "insert": "test",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#ff8700",
      },
      "insert": "^^^^",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "#ff5f00",
      },
      "insert": "test",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#ff5f00",
      },
      "insert": "++++",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "#ff0000",
      },
      "insert": "test",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#ff0000",
      },
      "insert": "####",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "#ffff00",
      },
      "insert": "= test",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#ffd700",
      },
      "insert": "== test",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#ffaf00",
      },
      "insert": "=== test",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#ff8700",
      },
      "insert": "==== test",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#ff5f00",
      },
      "insert": "===== test",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#ff0000",
      },
      "insert": "====== test",
    },
    Object {
      "insert": "


this is ",
    },
    Object {
      "attributes": Object {
        "color": "#0087ff",
      },
      "insert": "*bold*",
    },
    Object {
      "insert": "
this is ",
    },
    Object {
      "attributes": Object {
        "color": "#af00ff",
      },
      "insert": "_italic_",
    },
    Object {
      "insert": "
this is ",
    },
    Object {
      "attributes": Object {
        "color": "#8a8a8a",
      },
      "insert": "+mono+",
    },
    Object {
      "insert": " or ",
    },
    Object {
      "attributes": Object {
        "color": "#8a8a8a",
      },
      "insert": "\`mono\`",
    },
    Object {
      "insert": "
this is ",
    },
    Object {
      "attributes": Object {
        "color": "green",
      },
      "insert": "[underline]#underline#",
    },
    Object {
      "insert": "
this is ",
    },
    Object {
      "attributes": Object {
        "color": "maroon",
      },
      "insert": "[line-through]#strikethrough#",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "[",
    },
    Object {
      "attributes": Object {
        "color": "#7aa6da",
      },
      "insert": "red",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "]",
    },
    Object {
      "attributes": Object {
        "color": "",
      },
      "insert": "#this is a span of text#",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "[",
    },
    Object {
      "attributes": Object {
        "color": "#7aa6da",
      },
      "insert": "block",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "]",
    },
    Object {
      "insert": " with some other text ",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "[",
    },
    Object {
      "attributes": Object {
        "color": "#7aa6da",
      },
      "insert": "then another block",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "]",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "[[",
    },
    Object {
      "attributes": Object {
        "background": "white",
        "color": "#7aa6da",
      },
      "insert": "block id element",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "]]",
    },
    Object {
      "insert": "

Symbols: (R) (C) (V) (TM) -> <- => <= &#182; -- 

",
    },
    Object {
      "attributes": Object {
        "color": "#515a5a",
      },
      "insert": "// Keywords",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#cc9900",
      },
      "insert": "ifdef::",
    },
    Object {
      "insert": " ",
    },
    Object {
      "attributes": Object {
        "color": "#cc9900",
      },
      "insert": "ifndef::",
    },
    Object {
      "insert": " ",
    },
    Object {
      "attributes": Object {
        "color": "#cc9900",
      },
      "insert": "endif::",
    },
    Object {
      "insert": " ",
    },
    Object {
      "attributes": Object {
        "color": "#cc9900",
      },
      "insert": "ifeval::",
    },
    Object {
      "insert": " ",
    },
    Object {
      "attributes": Object {
        "color": "#cc9900",
      },
      "insert": "include::",
    },
    Object {
      "insert": " ",
    },
    Object {
      "attributes": Object {
        "color": "#cc9900",
      },
      "insert": "include1::",
    },
    Object {
      "insert": " ",
    },
    Object {
      "attributes": Object {
        "color": "#cc9900",
      },
      "insert": "sys::",
    },
    Object {
      "insert": " ",
    },
    Object {
      "attributes": Object {
        "color": "#cc9900",
      },
      "insert": "eval::",
    },
    Object {
      "insert": " ",
    },
    Object {
      "attributes": Object {
        "color": "#cc9900",
      },
      "insert": "template::",
    },
    Object {
      "insert": "


",
    },
    Object {
      "attributes": Object {
        "color": "#b9ca4a",
      },
      "insert": "http://www.google.com",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "[",
    },
    Object {
      "attributes": Object {
        "color": "#7aa6da",
      },
      "insert": "Google",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "]",
    },
    Object {
      "attributes": Object {
        "color": "#e7c547",
      },
      "insert": " \\"this is some more text\\"",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#b9ca4a",
      },
      "insert": "https://www.example.com",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "[",
    },
    Object {
      "attributes": Object {
        "color": "#7aa6da",
      },
      "insert": "Example",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "]",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#b9ca4a",
      },
      "insert": "ftp://www.example.com",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "[",
    },
    Object {
      "attributes": Object {
        "color": "#7aa6da",
      },
      "insert": "Example",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "]",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#b9ca4a",
      },
      "insert": "file:///www.example.com",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "[",
    },
    Object {
      "attributes": Object {
        "color": "#7aa6da",
      },
      "insert": "Example",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "]",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#b9ca4a",
      },
      "insert": "http://www.example.com",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "[",
    },
    Object {
      "attributes": Object {
        "color": "#7aa6da",
      },
      "insert": "Another Example",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "]",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "#b9ca4a",
      },
      "insert": "http://www.example.com",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "#b9ca4a",
      },
      "insert": "something@example.com",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#b9ca4a",
      },
      "insert": "mailto:joe.bloggs@foobar.com",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "[",
    },
    Object {
      "attributes": Object {
        "color": "#7aa6da",
      },
      "insert": "email Joe Bloggs",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "]",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "#b9ca4a",
      },
      "insert": "link:asciidoc",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "[",
    },
    Object {
      "attributes": Object {
        "color": "#7aa6da",
      },
      "insert": "This document",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "]",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#00ffff",
      },
      "insert": ".Thumbnail linked to full image",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#b9ca4a",
      },
      "insert": "image:/images/font/640-screen2.gif",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "[",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#7aa6da",
      },
      "insert": "\\"My screenshot\\",width=128,",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#7aa6da",
      },
      "insert": "link=\\"/images/font/640-screen2.gif\\"",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": "]",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "#00ffff",
      },
      "insert": ".Optional title",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "silver",
      },
      "insert": "[test configuration block for code]",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "silver",
      },
      "insert": "----------",
    },
    Object {
      "insert": "
let blah: number = \\"stuff\\";",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "

",
    },
    Object {
      "insert": "function test() {",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": "  if (test) {",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": "    for (let i = 0; i < 10; i++) {",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": "      console.log('blah');",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": "    }",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": "  }",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": "}",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "silver",
      },
      "insert": "----------",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "silver",
      },
      "insert": "-------------",
    },
    Object {
      "insert": "
 More Code",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "silver",
      },
      "insert": "-------------",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "silver",
      },
      "insert": "----------",
    },
    Object {
      "insert": "
Code",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "silver",
      },
      "insert": "----------",
    },
    Object {
      "insert": "


",
    },
    Object {
      "attributes": Object {
        "color": "silver",
      },
      "insert": "----------",
    },
    Object {
      "insert": "
SQL> describe aircraft;",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": " Name                   Null?  Type",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": " ----------------------------------------- -------- ----------------------------",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": " SERIAL_NO                 NOT NULL NUMBER(38)",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": " MODEL_NO                 NOT NULL VARCHAR2(15)",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "

",
    },
    Object {
      "insert": "SQL> describe flight;",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": " Name                   Null?  Type",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": " ----------------------------------------- -------- ----------------------------",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": " NUM                    NOT NULL NUMBER(38)",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": " ORIGIN                       VARCHAR2(3)",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": " DEST                        VARCHAR2(3)",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": " DEP_TIME                      VARCHAR2(5)",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "
",
    },
    Object {
      "insert": " ARR_TIME                      VARCHAR2(5)",
    },
    Object {
      "attributes": Object {
        "code-block": true,
      },
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "silver",
      },
      "insert": "----------",
    },
    Object {
      "insert": "


",
    },
    Object {
      "attributes": Object {
        "color": "#ffd700",
      },
      "insert": "test",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#ffd700",
      },
      "insert": "----------",
    },
    Object {
      "insert": "



",
    },
    Object {
      "attributes": Object {
        "color": "#515a5a",
      },
      "insert": "//////////////////////////////",
    },
    Object {
      "insert": "


",
    },
    Object {
      "attributes": Object {
        "color": "#515a5a",
      },
      "insert": "This is a comment block",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#515a5a",
      },
      "insert": "sdfsdf",
    },
    Object {
      "insert": "


",
    },
    Object {
      "attributes": Object {
        "color": "#515a5a",
      },
      "insert": "//////////////////////////////",
    },
    Object {
      "insert": "

between

",
    },
    Object {
      "attributes": Object {
        "color": "#515a5a",
      },
      "insert": "// This is a line comment",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#515a5a",
      },
      "insert": " // This is another line comment",
    },
    Object {
      "insert": "


",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": ". ",
    },
    Object {
      "insert": "slkdjfl
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": ".. ",
    },
    Object {
      "insert": "laskdjflkajsdlfkj
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "... ",
    },
    Object {
      "insert": "alskjdflaksjdf
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": ".... ",
    },
    Object {
      "insert": "asldkfjalksdjflkasjdf
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": ". ",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": ".. ",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "... ",
    },
    Object {
      "insert": "alsdjflaksjdflkajs

",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "- ",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "* ",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "** ",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "*** ",
    },
    Object {
      "insert": "

",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "1. ",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "2. ",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "3. ",
    },
    Object {
      "insert": "
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "4. ",
    },
    Object {
      "insert": "


",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "1. ",
    },
    Object {
      "insert": "alkjsdf lakjsd lfk

",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "- ",
    },
    Object {
      "insert": "alksdj flasjd flakjds f
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "  1. ",
    },
    Object {
      "insert": "laskjd flakjsd flkajd f

",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "- ",
    },
    Object {
      "insert": "slk ",
    },
    Object {
      "attributes": Object {
        "color": "#0087ff",
      },
      "insert": "*djflsk*",
    },
    Object {
      "insert": " jdf
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "  1. ",
    },
    Object {
      "insert": "alksdjflkasjdflkaj sdlkfj
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "    a. ",
    },
    Object {
      "insert": "alskdjflkajsdlfkja
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "    b. ",
    },
    Object {
      "insert": "alksjdflkajsdlkfjalsdkfj
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "  2. ",
    },
    Object {
      "insert": "alksdjflkasjdfkl
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "  3. ",
    },
    Object {
      "insert": "alskdj ",
    },
    Object {
      "attributes": Object {
        "color": "#af00ff",
      },
      "insert": "_flaksjdflkaj_",
    },
    Object {
      "insert": " sdfkljasldkfjlasdkfj
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "- ",
    },
    Object {
      "insert": "laskdj ",
    },
    Object {
      "attributes": Object {
        "color": "#8a8a8a",
      },
      "insert": "+flkasj+",
    },
    Object {
      "insert": " dflkaj
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "  * ",
    },
    Object {
      "insert": "alskdjflaksdjf
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "  ** ",
    },
    Object {
      "insert": "alksdjflaksdjfklajsdf
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "  ** ",
    },
    Object {
      "insert": "alksdjflaksdjf
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "  * ",
    },
    Object {
      "insert": "alksjdflkajsdfklj
",
    },
    Object {
      "attributes": Object {
        "color": "#e2b228",
      },
      "insert": "- ",
    },
    Object {
      "insert": "laksjdflaksdjflkajsdlfkjklajsdlfkjalsdkf

",
    },
    Object {
      "attributes": Object {
        "background": "red",
        "color": "white",
      },
      "insert": "TODO",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": ": ",
    },
    Object {
      "insert": "lorem ipsum
",
    },
    Object {
      "attributes": Object {
        "background": "red",
        "color": "white",
      },
      "insert": "FIXME",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": ": ",
    },
    Object {
      "insert": "lorem ipsum
",
    },
    Object {
      "attributes": Object {
        "background": "red",
        "color": "white",
      },
      "insert": "IMPORTANT",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": ": ",
    },
    Object {
      "insert": "lorem ipsum
",
    },
    Object {
      "attributes": Object {
        "background": "red",
        "color": "white",
      },
      "insert": "WARNING",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": ": ",
    },
    Object {
      "insert": "lorem ipsum
",
    },
    Object {
      "attributes": Object {
        "background": "red",
        "color": "white",
      },
      "insert": "TIP",
    },
    Object {
      "attributes": Object {
        "color": "gray",
      },
      "insert": ": ",
    },
    Object {
      "insert": "lorem ipsum

",
    },
  ],
}
`;

exports[`Use markup bold call with Asciidoc 1`] = `
Delta {
  "ops": Array [
    Object {
      "insert": "*test*
",
    },
  ],
}
`;

exports[`Use markup header 0 call with Asciidoc 1`] = `
Delta {
  "ops": Array [
    Object {
      "insert": "test
",
    },
  ],
}
`;

exports[`Use markup header 1 call with Asciidoc 1`] = `
Delta {
  "ops": Array [
    Object {
      "insert": "= test
",
    },
  ],
}
`;

exports[`Use markup header 2 call with Asciidoc 1`] = `
Delta {
  "ops": Array [
    Object {
      "insert": "== test
",
    },
  ],
}
`;

exports[`Use markup header 3 call with Asciidoc 1`] = `
Delta {
  "ops": Array [
    Object {
      "insert": "=== test
",
    },
  ],
}
`;

exports[`Use markup header 4 call with Asciidoc 1`] = `
Delta {
  "ops": Array [
    Object {
      "insert": "==== test
",
    },
  ],
}
`;

exports[`Use markup header 5 call with Asciidoc 1`] = `
Delta {
  "ops": Array [
    Object {
      "insert": "===== test
",
    },
  ],
}
`;

exports[`Use markup header 6 call with Asciidoc 1`] = `
Delta {
  "ops": Array [
    Object {
      "insert": "====== test
",
    },
  ],
}
`;

exports[`Use markup set call to change the mode to asciidoc 1`] = `
Object {
  "content": "test",
  "custom": Object {
    "background": "red",
    "foreground": "yellow",
  },
  "dirtyLimit": 20,
  "followLinks": false,
  "fontName": "Fira Code",
  "fontSize": 14,
  "highlight": "solarized-light",
  "idleDelay": 2000,
  "mode": 0,
  "onChange": [Function],
  "onClick": [Function],
  "onClickLink": [Function],
}
`;
